phoenix-core/src/main/java/org/apache/phoenix/coprocessor/UngroupedAggregateRegionObserver.java

```diff
@@ -588,6 +588,49 @@ public InternalScanner preCompact(ObserverContext<RegionCoprocessorEnvironment>
                 @Override
                 public InternalScanner run() throws Exception {
                     InternalScanner internalScanner = scanner;
+                    boolean isDisabled = false;
+                    if (request.isMajor()) {
+                        final String
+                                fullTableName =
+                                c.getEnvironment().getRegion().getRegionInfo().getTable().getNameAsString();
+                        if (!PhoenixDatabaseMetaData.SYSTEM_CATALOG_NAME.equals(fullTableName)) {
+                            try (PhoenixConnection conn = QueryUtil.getConnectionOnServer(
+                                    compactionConfig).unwrap(PhoenixConnection.class)) {
+                                PTable table = PhoenixRuntime.getTableNoCache(conn, fullTableName);
+                                List<PTable>
```

Comment:
We could also check what coprocessors the data table has, and only apply this logic if the old indexing is still used.

Reply:
Done